src/napari_spatialdata/_utils.py

```diff
@@ -99,20 +99,24 @@ def _get_categorical(
 def _position_cluster_labels(coords: NDArrayA, clusters: pd.Series, colors: NDArrayA) -> dict[str, NDArrayA]:
     if not is_categorical_dtype(clusters):
         raise TypeError(f"Expected `clusters` to be `categorical`, found `{infer_dtype(clusters)}`.")
-
     coords = coords[:, 1:]
     df = pd.DataFrame(coords)
     df["clusters"] = clusters.values
+    df["colors"] = np.array([to_hex(col) for col in colors])
+    colortypes = df.groupby(["clusters"])["colors"].apply(lambda g: list(np.unique(g))).reset_index(drop=True)
     df = df.groupby("clusters")[[0, 1]].apply(lambda g: list(np.median(g.values, axis=0)))
     df = pd.DataFrame(list(df), index=df.index)
     kdtree = KDTree(coords)
     clusters = np.full(len(coords), fill_value="", dtype=object)
     # index consists of the categories that need not be string
     clusters[kdtree.query(df.values)[1]] = df.index.astype(str)
+    order = np.argsort(kdtree.query(df.values)[1])
+    colortypes = colortypes.reindex(order)
+    colortypes = np.array([c[0] for c in colortypes])
     # napari v0.4.9 - properties must be 1-D in napari/layers/points/points.py:581
     colors = np.array([to_hex(col) for col in colors])
     colors = np.array([col if not len(cl) else to_hex((0, 0, 0)) for cl, col in zip(clusters, colors)])
```

Comment:
If I saw correctly `"colors"` was never used even before I changed anything. if that is really the case then maybe it would be useful to delete this part of the code and as an input of the function have the color pallette instead of the color vector of the points? (but maybe too many things would have to be changed)

Reply:
I think they are already removed in main